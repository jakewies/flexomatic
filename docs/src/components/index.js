export { default as Section } from './Section'
export { UnorderedList, ListItem } from './List'
export { Header1, Header2, TextItem, TextBlock, Code, Pre } from './Typography'
export { default as FancyLink } from './FancyLink'
export { default as Example } from './Example'
export { default as Page } from './Page'
