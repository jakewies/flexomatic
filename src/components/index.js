export { default as Grid } from './Grid'
export { default as Cell } from './Cell'
export { default as Demo } from './Demo'
