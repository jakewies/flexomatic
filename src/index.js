export { Grid, Cell, Demo } from './components'
