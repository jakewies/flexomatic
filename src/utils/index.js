export { default as align } from './align'
