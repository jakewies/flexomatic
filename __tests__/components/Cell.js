import React from 'react'
import { Cell } from '../../src/'
import renderer from 'react-test-renderer'
import 'jest-styled-components'

describe('Cell', () => {
  it('renders', () => {
    const tree = renderer.create(<Cell>Test Cell</Cell>).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders with proper width', () => {
    const tree = renderer.create(<Cell width={1 / 2}>Test</Cell>).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders with multiple widths', () => {
    const tree = renderer.create(<Cell width={[1, 'half', 0.25]}>Test</Cell>).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
