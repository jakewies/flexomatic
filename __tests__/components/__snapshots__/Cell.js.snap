// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cell renders 1`] = `
.c0 {
  box-sizing: border-box;
  display: block;
  width: auto;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

<div
  className="c0"
>
  Test Cell
</div>
`;

exports[`Cell renders with multiple widths 1`] = `
.c0 {
  box-sizing: border-box;
  display: block;
  width: 100%;
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
}

@media (min-width:768px) {
  .c0 {
    width: 50%;
  }
}

@media (min-width:1024px) {
  .c0 {
    width: 25%;
  }
}

<div
  className="c0"
  width={
    Array [
      1,
      "half",
      0.25,
    ]
  }
>
  Test
</div>
`;

exports[`Cell renders with proper width 1`] = `
.c0 {
  box-sizing: border-box;
  display: block;
  width: 50%;
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
}

<div
  className="c0"
  width={0.5}
>
  Test
</div>
`;
